app/src/main/java/gifapp/jsm/myapp/app/interview/jsmtech/gifapp/MainActivity.java

```diff
@@ -0,0 +1,185 @@
+package gifapp.jsm.myapp.app.interview.jsmtech.gifapp;
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.StaggeredGridLayoutManager;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.MotionEvent;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+
+import com.android.volley.Request;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+import adapter.GIFAdapter;
+import helper.EndlessRecyleOnScrollListenerStaggered;
+import helper.SingleTon;
+import helper.Utility;
+import model.GIFDetails;
+
+/**
+ * Created by Sukriti on 7/25/16.
+ */
+
+public class MainActivity extends AppCompatActivity {
+
+    private EditText mSearch;
+    private RecyclerView mRecycleView;
+    private ProgressBar mProgressBar;
+    private ArrayList<GIFDetails> listOfGifs;
+    private GIFAdapter gifAdapter;
+    private Integer offset = 0;
+    private String searchUrl;
+    private int calledMethod = 0;  // 0 -> Trending || 1 -> Search // othebe: Should use var names for this.
+    private Utility utility = new Utility();
+    String API_KEY = "dc6zaTOxFJmzC"; // Public beta key for the API // othebe: Should be final.
```

Comment:
Should be private final String, or be declared in a separate config class instead.